### DISCUSSION TITLE
Keycloak v23.0.3 Issue: Invalid Email Address Error During Password Reset with '+' Character

### DISCUSSION THREAD

**Discussion by adrianrosu05** [keycloak/keycloak#26066] [Q&A] [answered]
Greetings,

I'm currently experiencing an issue with Keycloak version 23.0.3 in my project. Initially, I created an user with the email address arosu1@gmail.com. After receiving the confirmation email from Keycloak, I successfully logged in.

Later, I created a new user using the email address arosu1+1@gmail.com. The sign-up and login processes were completed successfully. It's important to note that the sign-up confirmation emails are being sent to the arosu1@gmail.com address.

The problem arises when attempting to reset the password for the user with the email address arosu1+1@gmail.com using Keycloak's forgotten password API. Despite the email already existing in the system, I'm receiving the message: "Invalid email address." Interestingly, removing the "+" character allows the password reset to work as expected.

I would appreciate any advice or insights on this matter.

Thank you.

**Comment by dnwSilver** [comment]
Expected: duplicateEmailsAllowed=false compares full strings, so user@gmail.com and user+1@gmail.com are different in Keycloak.

keycloak-alias-guard canonicalizes the mailbox before uniqueness check (Gmail: dots, +tag, googlemail.com; others: +tag):
https://github.com/it-perspectives/keycloak-alias-guard

Validator id: alias-guard on the email attribute + event listener alias-guard-listener. Keycloak 26.x, declarative User Profile.

**Reply by ruchikajha95** [reply]
Hello @dnwSilver,

Thanks for the details! Since this is a really useful extension for enforcing true email uniqueness, it would be great to have it listed on the official Keycloak website so the community can find it easily.

The Keycloak website is managed out of the [keycloak/keycloak-web](https://github.com/keycloak/keycloak-web) repository. You can submit a Pull Request there to add keycloak-alias-guard to their official Extensions page.

For reference, you can look at similar PR like : https://github.com/keycloak/keycloak-web/pull/774 to see exactly how extensions are formatted.

Thanks

**Comment by ruchikajha95** [comment, marked as answer]
Hello @adrianrosu05 ,

Thank you for reaching out and providing such a clear description of the issue.

What you are experiencing is a known limitation/bug in older versions of Keycloak (such as v23.0.3).

We highly recommend upgrading your Keycloak instance to the latest stable release, Keycloak 26.6.

Please upgrade to Keycloak 26.6. If you encounter this specific issue or any other bugs on the latest version, please let us know, and we will be fully able to support you and investigate a fix.

Thanks.
